src/DataModel/IResource.ts

```diff
@@ -0,0 +1,5 @@
+export interface IResource
+{
+    "@id": string;
+    "@type"?: Array<string>
```

Comment:
Actually, [JSON-LD can do that](https://json-ld.org/playground/#startTab=tab-framed&json-ld=%7B%22%40id%22%3A%22http%3A%2F%2Fexample%2Fresource%22%2C%22http%3A%2F%2Ftest.property%22%3A%22s%22%7D&frame=%7B%22%40context%22%3A%7B%22id%22%3A%22%40id%22%7D%7D&context=%7B%22id%22%3A%22%40id%22%7D)